### PR TITLE
Add WatchListPageSize to cache.Config

### PR DESCRIPTION
**Pull request by mborsz**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Currently there is no way to specify WatchListPageSize used by Controller. This PR adds a field that can be used to specify this.

Context: https://github.com/kubernetes/perf-tests/pull/1441#issuecomment-683658227

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/assign @wojtek-t

### CODE REVIEW

**Comment by mborsz** [comment]
While I'm not sure if we will be able to use this change to fix issue in https://github.com/kubernetes/perf-tests/pull/1441#issuecomment-683658227 due to potential k/k release timelines, but I think it's a good change anyway (and we should be able to switch slo-monitor to it eventually).

File: staging/src/k8s.io/client-go/tools/cache/controller.go [hunk 1]
```diff
@@ -72,6 +72,9 @@ type Config struct {
 
 	// Called whenever the ListAndWatch drops the connection with an error.
 	WatchErrorHandler WatchErrorHandler
+	
+	// WatchListPageSize is the requested chunk size of initial and resync watch lists.
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/tools/cache/controller.go:76]: We use "resync" terminology in the context of resyncing from in-memory state. What you means is relist.

Comment (mborsz) [reply staging/src/k8s.io/client-go/tools/cache/controller.go]: I copied this line from reflector.go :D Fixed here. I can fix in reflector.go as well if you want.

**Pull request merged**